app/models/course.rb

```diff
@@ -312,6 +312,36 @@ def to_s
     "#{name} (#{course_code})"
   end
 
+  def copy_to_provider(new_provider)
+    new_course = new_provider
+                   .courses
+                   .find_by(course_code: self.course_code)
+    unless new_course
+      new_course = self.dup
+      new_provider.courses << new_course
+
+      if (last_enrichment = enrichments.latest_first.first)
+        new_enrichment = last_enrichment.dup
+        new_enrichment.draft! if new_enrichment.published?
+        # TODO: do we have to alter last_published_date ?
+        new_course.enrichments << new_enrichment
+      end
+
+      self.site_statuses.each do |site_status|
+        unless new_course.site_statuses.find_by site_id: site_status.site.id
+          new_site_status = site_status.dup
+          new_site_status.update(
+            vac_status: SiteStatus.default_vac_status_given(
+              study_mode: new_course.study_mode
+            ),
+            status: :new_status
```

Comment:
We need to edit `applications_accepted_from` too.

Reply:
Do we know what it gets set to?